cpp/tests/test_odesecirts.cpp

```diff
@@ -0,0 +1,1654 @@
+        model.parameters.get<mio::osecirts::ReducInfectedSevereCriticalDeadPartialImmunity<double>>()[i]  = 0.5;
+        model.parameters.get<mio::osecirts::ReducInfectedSevereCriticalDeadImprovedImmunity<double>>()[i] = 0.5;
+        model.parameters.get<mio::osecirts::ReducTimeInfectedMild<double>>()[i]                           = 1.0;
+    }
+
+    model.parameters.get<mio::osecirts::ICUCapacity<double>>()          = 10;
+    model.parameters.get<mio::osecirts::TestAndTraceCapacity<double>>() = 1.0;
+    const size_t daily_vaccinations                                     = 1;
+    const size_t num_days                                               = 5;
+    model.parameters.get<mio::osecirts::DailyPartialVaccinations<double>>().resize(mio::SimulationDay(num_days));
+    model.parameters.get<mio::osecirts::DailyFullVaccinations<double>>().resize(mio::SimulationDay(num_days));
+    model.parameters.get<mio::osecirts::DailyBoosterVaccinations<double>>().resize(mio::SimulationDay(num_days));
+    for (size_t i = 0; i < num_days; ++i) {
+        for (mio::AgeGroup j = 0; j < nb_groups; ++j) {
+            auto num_vaccinations = static_cast<double>(i * daily_vaccinations);
+            model.parameters.get<mio::osecirts::DailyPartialVaccinations<double>>()[{j, mio::SimulationDay(i)}] =
+                num_vaccinations;
+            model.parameters.get<mio::osecirts::DailyFullVaccinations<double>>()[{j, mio::SimulationDay(i)}] =
+                num_vaccinations;
+            model.parameters.get<mio::osecirts::DailyBoosterVaccinations<double>>()[{j, mio::SimulationDay(i)}] =
+                num_vaccinations;
+        }
+    }
+
+    mio::ContactMatrixGroup& contact_matrix = model.parameters.get<mio::osecirts::ContactPatterns<double>>();
+    const double cont_freq                  = 1;
+    const double fact                       = 1.0 / (double)(size_t)nb_groups;
+    contact_matrix[0] =
+        mio::ContactMatrix(Eigen::MatrixXd::Constant((size_t)nb_groups, (size_t)nb_groups, fact * cont_freq));
+    return model;
+}
+
+TEST(TestOdeSECIRTS, get_flows)
+{
+    auto& model = osecirts_testing_model();
+
+    Eigen::VectorXd y     = Eigen::VectorXd::Constant(model.get_initial_values().size(), 1.0);
+    Eigen::VectorXd flows = Eigen::VectorXd::Zero(model.get_initial_flows().size());
+
+    model.get_flows(y, y, 0.0, flows);
+
+    // // check flows individually
+    EXPECT_EQ(flows.size(), 53);
+
+    // Check each flow individually
+    EXPECT_EQ((flows[model.get_flat_flow_index<mio::osecirts::InfectionState::SusceptibleNaive,
+                                               mio::osecirts::InfectionState::ExposedNaive>({mio::AgeGroup(0)})]),
+              0.09375);
+
+    EXPECT_EQ((flows[model.get_flat_flow_index<mio::osecirts::InfectionState::SusceptibleNaive,
+                                               mio::osecirts::InfectionState::TemporaryImmunPartialImmunity>(
+                  {mio::AgeGroup(0)})]),
+              0.90625);
```

Comment:
This test could be shortened using a "compare" vector and a for loop to check all entries. The trade off is a more readable test that makes it easy to verify that all entries are checked, but it would be slightly harder to find out which flow failed a comparison.